integration/sawtooth_integration/tests/test_network_permissioning.py

```diff
@@ -0,0 +1,726 @@
+            self.clients.append(Client("http://127.0.0.1:8087"))
+
+        with open(os.path.join(self.sawtooth_home[1], 'keys', 'validator.pub'),
+                  'r') as infile:
+            non_genesis_key = infile.read().strip('\n')
+        with open(os.path.join(self.sawtooth_home[0], 'keys', 'validator.pub'),
+                  'r') as infile:
+            genesis_key = infile.read().strip('\n')
+
+        wait_for_consensus(self.clients, amount=2)
+
+        walter.set_pubkey_for_role(
+            "non_genesis_out_of_network",
+            "network.consensus",
+            permit_keys=[genesis_key],
+            deny_keys=[non_genesis_key])
+
+        wait_for_out_of_consensus(self.clients)
+        show_blocks(self.clients[0].block_list())
+        show_blocks(self.clients[1].block_list())
+
+        walter.set_pubkey_for_role(
+            "allow_all_for_consensus",
+            "network.consensus",
+            permit_keys=[genesis_key, non_genesis_key],
+            deny_keys=[])
+
+        wait_for_consensus(self.clients)
+        show_blocks(self.clients[0].block_list())
+        show_blocks(self.clients[1].block_list())
+
+def write_validator_config(sawtooth_home, **kwargs):
+    with open(os.path.join(sawtooth_home, 'etc',
+                           'validator.toml'), mode='w') as out:
+        toml.dump(kwargs, out)
+
+def start_validator(num, sawtooth_home):
+    return NodeController.start_node(
+        num,
+        NodeController.intkey_config_identity_registry,
+        NodeController.everyone_peers_with_everyone,
+        NodeController.even_parallel_odd_serial,
+        sawtooth_home,
+        NodeController.simple_validator_cmds,
+        {})
+
+
+def show_blocks(block_list):
+    blocks = [("Block Num", "Block ID", "Signer Key")] + block_list
+    output = "\n" + "\n".join(["{:^5} {:^21} {:^21}".format(item[0],
+                                            item[1][:10],
+                                            item[2][:10])
+               for item in blocks])
+    LOGGER.warning(output)
+
+
+class Client(object):
+
+    def __init__(self, rest_endpoint):
+        self.priv_key = signer.generate_privkey()
+        self.pub_key = signer.generate_pubkey(self.priv_key)
+        self._namespace = hashlib.sha512('intkey'.encode()).hexdigest()[:6]
+        self._factory = MessageFactory(
+            'application/cbor',
+            'intkey',
+            '1.0',
+            self._namespace)
+        self._rest = RestClient(rest_endpoint)
+
+
+    def send(self):
+        name = uuid4().hex[:20]
+        txns = [self._factory.create_transaction(
+            cbor.dumps({'Name': name,
+                        'Verb': 'set',
+                        'Value': 1000}),
+            inputs=[self._namespace + self._factory.sha512(name.encode())[-64:]],
+            outputs=[self._namespace + self._factory.sha512(name.encode())[-64:]],
+            deps=[])]
+        self._rest.send_batches(self._factory.create_batch(txns))
```

Comment:
In order to alleviate the test failure, perhaps these need to wait for commit

Reply:
send batch already uses return self._post('/batches?wait', batch_list). I think the issue is with the function that waits until the validators are out of consensus. Boyd said he would look at it tonight.